### PR TITLE
Rewrite to flat tree

### PR DESCRIPTION
**Pull request by skad0**
+ component was rewritten to use flat tree
+ TreeCheckbox bugs were fixed

### CODE REVIEW

File: src/Tree.js [hunk 1]
```diff
@@ -52,38 +23,70 @@ export default class Tree extends Component {
...
+        const result = tree;
+        // add all keys, except children to avoid data multiplying
+        result.push(this._processNode(node, parentId));
+
+        (node.children || []).forEach(child => this._buildTree(child, result, node.id));
+
+        return result;
+    }
+
+    _processNode(node, parentId) {
```
Comment (alex-zuy) [inline src/Tree.js:42]: please name this method `_buildFlatTreeNode`

File: src/Tree.js [hunk 2]
```diff
@@ -52,38 +23,70 @@ export default class Tree extends Component {
...
+        super(props);
+
+        this.state = {
+            flatTree: this._buildTree(props.nodes, [])
+        };
+
+        this._renderFlatNode = this._renderFlatNode.bind(this);
+    }
+
+    _buildTree(node, tree, parentId) {
```
Comment (alex-zuy) [inline src/Tree.js:32]: parameter names are misleading:
* `node` is actually tree
* `tree` is actually list of nodes

File: src/Tree.js [hunk 3]
```diff
@@ -52,38 +23,70 @@ export default class Tree extends Component {
...
+        const result = tree;
```
Comment (alex-zuy) [inline src/Tree.js:33]: I think this variable is pointless - it is just alias for method argument

**Review by alex-zuy — CHANGES_REQUESTED** [review]
Great work! In general flat structure seems to be much simpler to use

File: src/Tree.js (hunk 2 above)
Comment (skad0) [reply src/Tree.js:32]: thx, I'm going to move this code to separated entity and refactor it in this pr

**Comment by alex-zuy** [comment]
Thank you!

**Pull request merged**